### PR TITLE
create gift card component

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: src/components/GiftCard.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          <q-item-side icon="timer" />
+          <q-item-main label="Expired Date:" />
+          <q-item-side right>
+            <q-item-tile color="secondary">{{giftCard.createAt|calculatorExpiredDate(giftCard.expiry)}}</q-item-tile>
+          </q-item-side>
+        </q-item>
+      </q-list>
+    </div>
+    <div class="col-5">
+      <gen-qr-code dark="#000" :qrcode="giftCard.code" />
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: dark="#000"   -> same default ?

File: src/components/GiftCard.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          <q-item-main class="q-ml-none" label="Amount:" />
+          <q-item-side right>
+            <q-item-tile color="secondary">{{"$"+giftCard.amount}}</q-item-tile>
+          </q-item-side>
+        </q-item>
+        <q-item>
+          <q-item-side icon="timer" />
+          <q-item-main label="Expired Date:" />
+          <q-item-side right>
+            <q-item-tile color="secondary">{{giftCard.createAt|calculatorExpiredDate(giftCard.expiry)}}</q-item-tile>
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: calculatorExpiredDate -> calculateExpiredDate/getExpiredDate

File: src/components/GiftCard.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      type: Object,
+      default: () => {},
+    },
+  },
+  components: {
+    genQrCode,
+  },
+  filters: {
+    calculatorExpiredDate(val, expiry) {
+      let l10nEN = new Intl.DateTimeFormat('en-GB')
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: Intl ?

File: src/components/GiftCard.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      let createDate = new Date(val)
+      return l10nEN.format(new Date(createDate.setDate(createDate.getDate() + expiry)))
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: return l10nEN.format(new Date(createDate.setDate(createDate.getDate() + expiry)))

-> why so complicated ?

File: src/components/GiftCard.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    </div>
+  </div>
+</template>
+<script>
+import genQrCode from './GenQRCode'
+export default {
+  props: {
+    giftCard: {
+      type: Object,
+      default: () => {},
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: default to empty object, what so giftCard.amount could be undefined ? and so is giftCard.expiry

File: src/pages/Stores.vue [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<template>
```
Comment (hemvanh) [inline src/pages/Stores.vue]: Stores.vue -> pls exclude testing pages

File: CHANGELOG.md [hunk 7]
```diff
@@ -54,4 +54,5 @@ All notable changes to this project will be documented in this file.
 - Logins fields validation
 - Login admin
 - Fetch admin -> fetch Admin info (username) and show to admin default logged in page
-- Creat seeders GiftCard add amount, expiry
\ No newline at end of file
+- Creat seeders GiftCard add amount, expiry
+- Gift card component
```
Comment (hemvanh) [inline CHANGELOG.md]: Gift card component -> A single GiftCard component used for printing

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/GiftCard.vue (hunk 1 above)
Comment (tri-et) [reply src/components/GiftCard.vue]: default is #00F(blue)

File: src/components/GiftCard.vue [hunk 8]
```diff
@@ -27,21 +27,18 @@
...
 <script>
 import genQrCode from './GenQRCode'
+import {date} from 'quasar'
 export default {
   props: {
-    giftCard: {
-      type: Object,
-      default: () => {},
-    },
+    giftCard: [Object],
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: this component is obviously a GiftCard, so the props for the GiftCard should be:
code, amount, **createdAt** and expiry which all have a default value
Dont create a GiftCard component with a single GiftCard prop

File: src/components/GiftCard.vue [hunk 9]
```diff
@@ -15,7 +15,7 @@
           <q-item-side icon="timer" />
           <q-item-main label="Expired Date:" />
           <q-item-side right>
-            <q-item-tile color="secondary">{{giftCard.createAt|calculatorExpiredDate(giftCard.expiry)}}</q-item-tile>
+            <q-item-tile color="secondary">{{giftCard.createAt|calculateExpiredDate(giftCard.expiry)}}</q-item-tile>
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: createAt -> createdAt

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/GiftCard.vue [hunk 10]
```diff
@@ -30,7 +30,22 @@ import genQrCode from './GenQRCode'
...
+      type: Number,
+      default: 0,
+    },
+    expiry: {
+      type: Number,
+      default: 0,
+    },
+    createdDate: {
+      type: String,
+      default: new Date().toDateString(),
```
Comment (hemvanh) [inline src/components/GiftCard.vue]: createdDate should not be a String, can choose between (timestamp (Number) getTime() or a Date object)
Date stored as String with no formatting rules wont be stable
----> store as String will have unnecessary convert back and forth between date and string

**Review by hemvanh — CHANGES_REQUESTED** [review]

**Review by hemvanh — APPROVED** [review]

**Pull request merged**